packages/graphql-hooks/test-node/GraphQLClient.test.ts

```diff
@@ -162,8 +162,113 @@ describe('with files in Node JS', () => {
+    const resultMock = { hello: 'World' }
+    const updaterMock = jest.fn(() => resultMock)
+
+    client.getCacheKey = getCacheKeyMock
+    client.cache = cacheMock
+    client.mutationsEmitter = mutationsEmitterMock
+
+    client.setQueryData(validQuery, updaterMock, operation)
+    expect(getCacheKeyMock).toBeCalledTimes(1)
+    expect(getCacheKeyMock).toBeCalledWith({ query: validQuery, ...operation })
+
+    operation.operationName = 'Operation2'
+
+    client.setQueryData(validQuery, updaterMock, operation)
+
+    expect(getCacheKeyMock).toBeCalledTimes(2)
+    expect(getCacheKeyMock).toBeCalledWith({ query: validQuery, ...operation })
+
+    expect(cacheMock.get).toBeCalledTimes(2)
+    expect(cacheMock.get).toBeCalledWith({ query: validQuery, ...operation })
+
+    expect(updaterMock).toBeCalledTimes(2)
+    expect(updaterMock).toBeCalledWith(resultMock)
+
+    expect(mutationsEmitterMock.emit).toBeCalledTimes(2)
+    expect(mutationsEmitterMock.emit).toBeCalledWith(Events.DATA_UPDATED, {
+      data: resultMock
+    })
+  })
+
+  it('correctly sets query data if the operation is null or undefined', () => {
+    const client = new GraphQLClient(validConfig)
+    const stateMock = { hello: 'World' }
+    const olderStatateMock = { data: stateMock }
+    const mutationsEmitterMock = {
+      ...new EventEmitter(),
+      emit: jest.fn()
+    }
+    const cacheMock = {
+      get: jest.fn(() => olderStatateMock),
+      set: jest.fn(),
+      delete: jest.fn(),
+      clear: jest.fn(),
+      keys: jest.fn(),
+      getInitialState: jest.fn()
+    }
+    const getCacheKeyMock = jest.fn(query => query)
+    const resultMock = { hello: 'World' }
+    const updaterMock = jest.fn(() => resultMock)
+
+    client.getCacheKey = getCacheKeyMock
+    client.cache = cacheMock
+    client.mutationsEmitter = mutationsEmitterMock
+
+    client.setQueryData(validQuery, updaterMock)
+    expect(getCacheKeyMock).toBeCalledTimes(1)
+    expect(getCacheKeyMock).toBeCalledWith({ query: validQuery })
+
+    expect(cacheMock.get).toBeCalledTimes(1)
+    expect(cacheMock.get).toBeCalledWith({ query: validQuery })
+
+    expect(updaterMock).toBeCalledTimes(1)
+    expect(updaterMock).toBeCalledWith(resultMock)
+
+    expect(mutationsEmitterMock.emit).toBeCalledTimes(1)
+    expect(mutationsEmitterMock.emit).toBeCalledWith(Events.DATA_UPDATED, {
+      data: resultMock
+    })
+
+    client.setQueryData(validQuery, updaterMock, null)
+
+    expect(getCacheKeyMock).toBeCalledTimes(2)
+    expect(getCacheKeyMock).toBeCalledWith({ query: validQuery })
+
+    expect(cacheMock.get).toBeCalledTimes(2)
+    expect(cacheMock.get).toBeCalledWith({ query: validQuery })
+
+    expect(updaterMock).toBeCalledTimes(2)
+    expect(updaterMock).toBeCalledWith(resultMock)
+
+    expect(mutationsEmitterMock.emit).toBeCalledTimes(2)
+    expect(mutationsEmitterMock.emit).toBeCalledWith(Events.DATA_UPDATED, {
+      data: resultMock
+    })
+  })
```

Comment:
Maybe you should add one more test using `variables` property

Reply:
Done